nntrainer/layers/layer_internal.h

```diff
@@ -301,6 +311,29 @@ class Layer : public ml::train::Layer {
     output_dim.resize(num_outputs);
   }
 
+  std::vector<Tensor> getHidden();
+
+  std::vector<Tensor> getGradient();
+
+  void resizeNetInput(unsigned int size){net_input.resize(size);}
```

Comment:
do all these netInput/netOutput related functions need to be public? I think this information is available at via properties and be set at initialize(). For example, resizeNetInput() is same as setProperty("num_inputs=..");

Reply:
This is only for the testing. We made unittest cases layer by layer. not based on graph. If we are using graph, then we don't need. graph will set the net_input at compile time. However, we need to set somehow if we are using just for the layer. 
these are same for the
``` resizeNetInput, resizeNetOutput, getNumInputs, getNumOutputs, setInputBuffer, setOutputBuffer ```
we need to enable these func for the test only.